src/components/AppHeader/AppSwitcher.js

```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+import {
+ Dropdown, Menu, Button, Icon,
+} from 'antd';
+import { withRouter } from 'react-router-dom';
+import { connect } from 'react-redux';
+
+import { setCurrentApp } from '../../batteries/modules/actions';
+
+const AppSwitcher = ({
+ apps, currentApp, history, updateCurrentApp,
+}) => {
+	const menu = (
+		<Menu
+			style={{ maxHeight: 250, overflowY: 'scroll' }}
+			onClick={(e) => {
+				const appName = e.key;
+				updateCurrentApp(appName, apps[appName]);
+				history.push(`/app/${appName}`);
```

Comment:
This causes the app to lose the active view and always redirects the current view to the overview page.

Reply:
Like if we are on search preview page and when we switch we want the same page to open for other app?
Am I right?

Reply:
yes

Reply:
Sounds good.